oak/server/rust/oak_runtime/src/node/mod.rs

```diff
@@ -42,6 +45,8 @@ pub trait Node: Send {
         handle: oak_abi::Handle,
         notify_receiver: oneshot::Receiver<()>,
     );
+
+    fn get_privilege(&self) -> NodePrivilege;
```

Comment:
Why not have a [default implementation](https://doc.rust-lang.org/book/ch10-02-traits.html#default-implementations) for this method and lose the changes to most Node types?

Reply:
Also, at the moment having this method in the trait (and so required for all Node types) seems awkward:
 - Only the Wasm node populates the method, all others fall back to the default.
 - The value returned for gRPC client and pseudo-Nodes will necessarily be misleading and wrong, because the type of the return value can't express their privilege (no wildcard `can_declassify: "user:*"` / `can_declassify: "domain:*"`).

But maybe the future plans for this method would obviate some or all of that?

Reply:
Re: default implementation, I am not a big fan (it feels too OOP for my taste), but I've implemented it anyways now.

I don't have a concrete plan for gRPC nodes, but I think we will need to refactor things around to make it work for them.

For gRPC client node, since the domain is specified as part of the creation, it should be possible for the node to just determine its own privilege at start of day, if we only allow exact matching on domains (i.e. the node would get the node to declassify exactly one tag, containing the full domain, or SNI, or whatever we decide). If we wanted something more sophisticated (e.g. partial matching), then I think we need to expand the "flows to" relationship between labels, to account for the fact that, e.g. "*.google.com" flows to "mail.google.com" etc., basically construct a lattice around these individual labels.

The gRPC server node is even trickier: it would be tricky for a single gRPC node to keep changing its label based on the user connecting to it (#1068 only sets the label on the invocation channels, but not on the node itself). I think a more scalable approach would be for the gRPC server node to spawn gRPC server per-invocation worker nodes, each with a fixed label corresponding to the user and to the label passed over gRPC metadata. This was also suggested by @aferr , in order to facilitate reasoning about non-interference and downgrades separately.